File: emulinker/src/main/java/org/emulinker/kaillera/controller/v086/action/V086Action.java
package org.emulinker.kaillera.controller.v086.action;

import org.emulinker.kaillera.controller.v086.V086Controller.V086ClientHandler;
import org.emulinker.kaillera.controller.v086.protocol.V086Message;

public interface V086Action<T extends V086Message> {
  @Override
  public String toString();

  public void performAction(T message, V086ClientHandler clientHandler) throws FatalActionException;

  public int getActionPerformedCount();
}

File: emulinker/src/main/java/org/emulinker/kaillera/controller/v086/action/V086Action.kt
package org.emulinker.kaillera.controller.v086.action

import kotlin.Throws
import org.emulinker.kaillera.controller.v086.V086ClientHandler
import org.emulinker.kaillera.controller.v086.protocol.V086Message

interface V086Action<T : V086Message> {
  override fun toString(): String

  @Throws(FatalActionException::class)
  fun performAction(message: T, clientHandler: V086ClientHandler?)
  val actionPerformedCount: Int
}
